### PR TITLE
Model - Validation 🌵 

### PR DESCRIPTION
**Pull request by evansnyamekye**
### Features:

Add the following validations: 
- [ ] For User `model`:
- The name must not be blank 🃏.
- PostCounter must be an integer greater than or equal to zero 🎲.
- [ ] For Post `model`:
- The title must not exceed 260.
- CommentsCounter must be an integer greater than or equal to zero 👍 . 
- LikesCounter must be an integer greater than or equal to zero 🔢 . 
- Add unit spec for all of your model methods and validation ✌️.

### CODE REVIEW

File: Gemfile
Comment (shahadat3669) [inline Gemfile:1]: 👍 Great progress on your test cases! 👍 However, there's a crucial step that needs attention to ensure the test environment is set up correctly:

- [ ] **Adding `factory_bot_rails` to Gemfile:** While your test cases appear to satisfy the requirements mostly, I noticed that the `factory_bot_rails` gem is not added to your `Gemfile`. So I was unable to run those test cases. To rectify this, please add the gem to your `Gemfile` and configure it appropriately. 🙏 

By including the gem in your project and configuring it correctly, you'll be able to run your test cases smoothly. Keep up the good work! 🚀🛠️

**Review by shahadat3669 — CHANGES_REQUESTED** [review]
Hi @evansnyamekye, 

Great job so far! 👏 You've done well, and you're almost there for the next project ⏭️, but there are a few issues 🐛 that you still need to work on.




## Don't Worry 🙏

It's okay to have bugs in your code. 🆗 Please see this code review as an opportunity to learn practical coding skills and discover new tips that can improve your code quality 💯 and help you **get even better** 💼.

**I will guide you on how to address these areas down below 👇.**

## Highlights
✅ `User` model has validations for name and post count
✅ `Post` model has validations for title, comments count and LikesCounter
✅ `Post` model has validations for "Title must not exceed 250 characters"


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Shahadat Hossain</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/shahadat3669">
        GitHub
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/shahadat-cseng/">
        LinkedIn
      </a>
    </td>
    <td align="center">
      <a href="https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V">
        Slack
      </a>
    </td>
    <td align="center">
      <a href="mailto:shahadat3669@gmail.com?subject=Microverse%20Code%20Review">
        Email
      </a>
    </td>
  </tr>
</table>

## ⚠️ WARNING ⚠️

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: Gemfile
Comment (evansnyamekye) [reply Gemfile:1]: Hi @shahadat3669 . Thank you for your review. All corrections and issues are resolved as requested by the reviewer.

Comment (Efiamotu-1) [inline Gemfile:1]: - [ ] Good job adding the chnages from the previous reviewer, however with the changes added your codebase no longer runs the tests as it is supposed to, so i kindly suggest you look into this, fix it and resubmit for approval :+1: 
ere

**Review by Efiamotu-1 — CHANGES_REQUESTED** [review]
Hi @evansnyamekye, 



Great work on making the changes requested by a previous reviewer 👏 Your Code is Awesome! 💪
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

There are some **few** issues 🐛 that you still need to work 🔧 on though to go to the next project ⏭️ but you are almost there! 🏃
## Don't Worry 🙏

It's ok to have bugs in your code 🐛. Please, see this code review as a **chance** for you to **learn** practical coding skills and learn **new tips** that can improve your code quality 💯 and help you **get a better ** 💼.

_I am going to guide you on how to fix them down below. 🤓_

## Highlights :dart: 
- ✔️ No linter Errors
- ✔️ Descriptive README
- ✔️ Descriptive PR title and description
- The code is top-notch! 💯 and has a huge potential.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

> **Note:** Please, don't Merge This Pull Request yet. Only Merge when it is Approved ✅.


## Thank you for your time 🙏 and call me if you need help or clarification ☎️

- This is My **GitHub**: [@Efiamotu-1](https://github.com/Efiamotu-1/)
- This is My **linkedIn**.[Habeeb Efiamotu Musa](https://www.linkedin.com/in/musa-habeeb/)
- This is My **Twitter**: [@EFYAMOTU](https://twitter.com/EFYAMOTU)
- This is My **Slack**: [Habeeb Efiamotu Musa](https://slack.com)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by binhussen — APPROVED** [review]
Hi @evansnyamekye , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉
IApproveSimonCowellGIF (6)

## Highlights
- Added Validation for Models ✔️
- Well Stractured Code ✔️
- Used Git flow ✔️
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**